### DISCUSSION TITLE
Is it possible to read traits for a specific Noun on-chain?

### DISCUSSION THREAD

**Discussion by cdt-eth** [nounsDAO/nouns-monorepo#409] [Solidity / smart contracts] [answered]
Is it possible to read traits for a specific Noun on-chain?

**Comment by solimander** [comment, marked as answer]
Yes, Noun trait seeds can be fetched on-chain by calling the `seeds(nounId)` function on the [NounsToken](https://etherscan.io/address/0x9c8ff314c9bc7f6e59a9d9225fb22946427edc03) contract. The returned seed contains an array index for each trait, which can be used to [lookup the trait name and image data](https://github.com/nounsDAO/nouns-monorepo/blob/master/packages/nouns-assets/src/utils.ts#L12-L22).
